### PR TITLE
[NBS Docs] Update adding resource permission check tutorial

### PR DESCRIPTION
**Pull request by camilaibs**
Signed-off-by: Camila Belo <camilaibs@gmail.com>

## Hey, I just made a Pull Request!

Closes: https://github.com/backstage/backstage/issues/26364

Update the  adding resource permission check tutorial to use the new backend system.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md [hunk 1]
```diff
@@ -80,9 +82,10 @@ router.put('/todos', async (req, res) => {
   const decision = (
     await permissions.authorize(
       [{ permission: todoListUpdatePermission, resourceRef: req.body.id }],
-      {
-        token,
-      },
+      await auth.getPluginRequestToken({
+        onBehalfOf: credentials,
+        targetPluginId: 'permission',
+      }),
```
Comment (Rugvip) [inline docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md:88]: credentials should be forwarded here instead, the permissions service no longer accepts a token

File: docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md [hunk 2]
```diff
@@ -80,9 +82,7 @@ router.put('/todos', async (req, res) => {
   const decision = (
     await permissions.authorize(
       [{ permission: todoListUpdatePermission, resourceRef: req.body.id }],
-      {
-        token,
-      },
+      credentials,
```
Comment (Rugvip) [inline docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md:85]:
```suggestion
      { credentials },
```

Still an options object 😅

Comment (camilaibs) [reply docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md:85]: That is interesting, I don't get any type error, even before when I was passing the token...

Comment (camilaibs) [reply docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md:85]: But if I apply the suggestion I get this error:
image

Comment (Rugvip) [reply docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md:85]: hmm, are you perhaps testing with the client-side permissions API? Is the `permissions` client you got the implementing `PermissionsService`?

Comment (camilaibs) [reply docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md:85]: I'm using `coreServices.permissions`, is that right 🤔 ?

Comment (camilaibs) [reply docs/permissions/plugin-authors/03-adding-a-resource-permission-check.md]: I was using a wrong type in the router options, it should be fixed now 🙏🏻 .

**Review by Rugvip — APPROVED** [review]
:shipit: 🎉

**Pull request merged**